src/test/java/wolox/training/controllers/BookControllerTest.java

```diff
@@ -0,0 +1,94 @@
+package wolox.training.controllers;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.mockito.ArgumentMatchers.any;
+
+import java.util.Arrays;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+
+
+@RunWith(SpringRunner.class)
+@AutoConfigureMockMvc
+@WebMvcTest(BookController.class)
+@AutoConfigureTestDatabase(replace = Replace.NONE)
+public class BookControllerTest extends BaseControllerTest {
+
+  @MockBean
+  private BookRepository mockBookRepository;
+  private Book book = new Book();
+  private String bookAsJson = "{\"id\":1,\"author\":\"Author\",\"image\":\"La imagen\",\"title\":\"El título\",\"subtitle\":\"El subtítulo\",\"publisher\":\"El publisher\",\"year\":\"1989\",\"pages\":250,\"isbn\":\"ISBN\",\"users\":[]}";
```

Comment:
I don't like this at all. Is there a better way? I don't want to copy-paste this string everywhere, I guess I could make it dynamically? Using Gson or something?

Reply:
There is `org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath` that could help you with this :) 
Here is a [link](https://www.baeldung.com/integration-testing-in-spring#2-verify-response-body) that could be useful